src/twisted/mail/test/test_smtp.py

```diff
@@ -1639,6 +1650,59 @@ def test_requireTLSAndHELOFallbackSucceedsIfOverTLS(self):
 
 
 
+class SMTPUTF8Option(unittest.TestCase):
+    """
+    The RFC 6521 "SMTPUTF8" option on the server specifies support for
+    UTF-8 encoded mail headers.  The client should not send
+    UTF-8 encoded bytes to a server which does not advertise the "SMTPUTF8"
+    option.
+    """
+    def setUp(self):
+        self.clientProtocol = smtp.ESMTPClient(
+            b"testpassword", None, b"testuser")
+        transport = StringTransport()
+        self.clientProtocol.requireAuthentication = False
+        self.clientProtocol.requireTransportSecurity = True
+        self.clientProtocol.heloFallback = True
+        self.clientProtocol.makeConnection(transport)
+        self.clientProtocol.dataReceived(b"220 localhost\r\n")
+        transport.clear()
+
+
+    def test_serverLacksSMTPUTF8(self):
+        """
+        If there is no SMTPUTF8 option offered by the server,
+        the client switches encoding to ASCII.
+        """
+        self.clientProtocol.dataReceived(
+            b"250-AUTH LOGIN\r\n"
+            b"250 STARTTLS\r\n")
+        self.assertEqual(self.clientProtocol._encoding, "ascii")
+
+        # Try to send UTF-8 text when client encoding is ascii
+        # will result in an error which causes the transport to disconnect.
+        self.clientProtocol.sendLine(u"RCPT TO:<прив@example.com>".encode("utf-8"))
+        self.assertTrue(self.clientProtocol.transport.disconnecting)
```

Comment:
Is it possible to assert that an exception is raised?

Reply:
I couldn't figure out how to get the exception here, but I found that self.clientProtocol.transport.disconnecting is set to True on failure.